js/src/BuilderSettingsMenu.js

```diff
@@ -0,0 +1,409 @@
+/**
+ * SettingsMenu
+ */
+
+/** @jsx h */
+import { h, Component } from 'preact'
+import '../../css/src/SettingsMenu.css'
+
+const ScaleEditor = require('./ScaleEditor')
+
+class BuilderSettingsMenu extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      display: props.display
+    }
+    props.settings.hold_changes()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({display: nextProps.display})
+    nextProps.settings.hold_changes()
+  }
+
+  abandonChanges () {
+    this.props.settings.abandon_changes()
+    this.setState({display: 'none'})
+  }
+
+  saveChanges () {
+    this.props.settings.accept_changes()
+    this.setState({display: 'none'})
+  }
+
+  handleStyle (value, type) {
+    let style = []
+    if (type === 'reaction_styles') {
+      style.concat(this.props.reaction_styles)
+    } else if (type === 'metabolite_styles') {
+      style.concat(this.props.metabolite_styles)
+    }
+    if (style) {
```

Comment:
An empty array is truthy `!![] === true`, you may want to assert on its length `!![].length === false`.

    if (style.length) {

Reply:
Since the conditions above are mutually exclusive, alternatively you can do
```
const style = type === 'reaction_styles' ? this.props.reaction_styles
  : type === 'metabolite_styles' ? this.props.metabolite_styles
  : null
if (style) {
```
I think the ternary operator is extremely useful, but it depends on the convention if it's okay to use it.

Reply:
This code is no longer used